apollo-ios/Design/swift-identifiable.md

```diff
@@ -1,29 +1,47 @@
 # Summary
 
-This document proposes a new GraphQL directive named `@apollo_client_ios_identity`, to be used to mark a field which can uniquely identify an object.
+This document proposes a new GraphQL directive named `@identity`, to be used to mark a field which can uniquely identify an object.
 
 # Introduction
 
 SwiftUI makes heavy use of the [Identifiable](https://developer.apple.com/documentation/swift/identifiable) protocol, which is used to track the identity of entities across data changes. If an object with a List is replaced with a different object with the same identity, SwiftUI will animate the item changing instead of animating an insertion. Objects that do not conform to the Identifiable protocol require additional boilerplate to be usable inside SwiftUI.
 
 Apollo Client iOS could assist the developer by adding conformance to the Identifiable protocol to its generated models. Selecting the field to be used as an identity is done by adding a new GraphQL directive.
 
+A concept of identity is required to allow response objects to be cached. The various Apollo Client projects have mechanisms that allow for identifiers to be selected through additional code. The new directive could allow schema authors to assist client authors with caching.
```

Comment:
> The new directive could allow schema authors to assist client authors with caching.

This is a nice ideal but not likely, as you noted below in [Usage in operations](https://github.com/apollographql/apollo-ios-dev/pull/549/commits/6430f37b1cc2931dd5236c4087a78e31a3ecffb8#diff-acc259b506e96644229b6d8785740a9248ed3849714b716e6b354258fe712004R63).
* Federated schema authors already have the [`@key` directive](https://www.apollographql.com/docs/graphos/reference/federation/directives#key) but I don't think those are passed through in the API schema SDL, nor introspection.
* You'll end up with a string of different directives if there isn't uniformity across all the clients.

Reply:
I agree this document is coming close to a [xkcd 927 situation](https://xkcd.com/927/). I wasn't aware of the `@key` directive, but it not being revealed to the clients is a deal-breaker for my use case.

Reply:
We have not previously defined any directives that are exposed as part of the public schema and are intended for consumption by the Apollo client's specifically. While this is certainly possible, we need to consider all of the issues this can cause. These directives would need to be namespaced accordingly. It is very possible that some schema's out there already have an `@identity` directive that they are using for some other use case. We can't just assume that directive is what we are looking for and consume it on the client for this usage.

Namespacing of directives doesn't really have great support in the GraphQL spec currently. We'd need to use some ugly directive name like `@apollo_client_identity`. I know that an [`@core` directive ](https://github.com/apollographql/specs/blob/main/core/v0.2/core-v0.2.md) was discussed by us at some point to help with this, but it's not an accepted and used feature currently. @martinbonnin knows more about that than I do.